### ISSUE TITLE
Incorrect private key password results in uncaught error rather than prompt for correct passphrase when using SFTP

### ISSUE THREAD

**Issue by brentpayton** [icetee/remote-ftp#516]
[Enter steps to reproduce below:]
1. Configure .ftpconfig to . . . 
   a)  use sftp
   b)  point to a password-protected openssh private key
   c)  not have a password
2. Attempt to connect to the server an provide an incorrect password.

[Discussion]

I think the text of the uncaught error is sufficient to let the user know they typed the wrong password.

Detecting and warning if caps lock is on would be nice.  If not feasible, prompt the user to check caps lock.

[Error log from Atom]

**Atom Version**: 1.11.2
**Electron Version**: 0.37.8
**System**: Microsoft Windows 10 Pro
**Thrown From**: [Remote-FTP](https://github.com/mgrenier/remote-ftp) package, v0.9.0
### Stack Trace

Uncaught Error: Malformed private key (expected sequence). Bad passphrase?

```
At C:\Users\brent\.atom\packages\Remote-FTP\node_modules\ssh2-streams\lib\utils.js:322

Error: Malformed private key (expected sequence). Bad passphrase?
    at genPublicKey (C:\Users\brent\.atom\packages\Remote-FTP\node_modules\ssh2-streams\lib\utils.js:322:13)
    at Client.connect (C:\Users\brent\.atom\packages\Remote-FTP\node_modules\ssh2\lib\client.js:155:29)
    at ConnectorSFTP.module.exports.ConnectorSFTP.connect (C:\Users\brent\.atom\packages\Remote-FTP\lib\connectors\sftp.js:78:13)
    at Client.module.exports.Client.doConnect (C:\Users\brent\.atom\packages\Remote-FTP\lib\client.js:297:18)
    at space-pen-div.<anonymous> (C:\Users\brent\.atom\packages\Remote-FTP\lib\client.js:719:9)
    at space-pen-div.jQuery.event.dispatch (C:\Users\brent\.atom\packages\Remote-FTP\node_modules\jquery\dist\jquery.js:4435:9)
    at space-pen-div.elemData.handle (C:\Users\brent\.atom\packages\Remote-FTP\node_modules\jquery\dist\jquery.js:4121:28)
    at Object.jQuery.event.trigger (C:\Users\brent\.atom\packages\Remote-FTP\node_modules\jquery\dist\jquery.js:4350:12)
    at space-pen-div.<anonymous> (C:\Users\brent\.atom\packages\Remote-FTP\node_modules\jquery\dist\jquery.js:4901:17)
    at Function.jQuery.extend.each (C:\Users\brent\.atom\packages\Remote-FTP\node_modules\jquery\dist\jquery.js:374:23)
    at PromptPassDialog.jQuery.fn.jQuery.each (C:\Users\brent\.atom\packages\Remote-FTP\node_modules\jquery\dist\jquery.js:139:17)
    at PromptPassDialog.jQuery.fn.extend.trigger [as originalTrigger] (C:\Users\brent\.atom\packages\Remote-FTP\node_modules\jquery\dist\jquery.js:4900:15)
    at PromptPassDialog.$.fn.trigger (C:\Users\brent\.atom\packages\Remote-FTP\node_modules\space-pen\lib\space-pen.js:606:19)
    at PromptPassDialog.module.exports.PromptPassDialog.onConfirm (C:\Users\brent\.atom\packages\Remote-FTP\lib\dialogs\prompt-pass-dialog.js:50:8)
    at space-pen-div.PromptPassDialog.atom.commands.add.core:confirm (C:\Users\brent\.atom\packages\Remote-FTP\lib\dialogs\prompt-pass-dialog.js:44:10)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (C:\Users\brent\AppData\Local\atom\app-1.11.2\resources\app.asar\src\command-registry.js:260:29)
    at C:\Users\brent\AppData\Local\atom\app-1.11.2\resources\app.asar\src\command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (C:\Users\brent\AppData\Local\atom\app-1.11.2\resources\app.asar\node_modules\atom-keymap\lib\keymap-manager.js:580:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (C:\Users\brent\AppData\Local\atom\app-1.11.2\resources\app.asar\node_modules\atom-keymap\lib\keymap-manager.js:388:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (C:\Users\brent\AppData\Local\atom\app-1.11.2\resources\app.asar\src\window-event-handler.js:114:36)
    at HTMLDocument.<anonymous> (C:\Users\brent\AppData\Local\atom\app-1.11.2\resources\app.asar\src\window-event-handler.js:3:61)
```
### Commands

```
     -3:55.1.0 editor:consolidate-selections (atom-text-editor.editor.mini.is-focused)
     -3:55.1.0 core:cancel (atom-text-editor.editor.mini.is-focused)
  2x -3:48.6.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -2:41 application:open-file (atom-pane.pane.active)
     -2:37.7.0 application:open-folder (atom-pane.pane.active)
     -2:31.6.0 command-palette:toggle (atom-pane.pane.active)
     -2:28.1.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -2:27.8.0 core:move-up (atom-text-editor.editor.mini.is-focused)
     -2:27.4.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -2:27.4.0 settings-view:core (atom-pane.pane.active)
     -0:17.2.0 command-palette:toggle (ul.list-inline.tab-bar.inset-panel)
  9x -0:13.4.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -0:11.5.0 core:move-up (atom-text-editor.editor.mini.is-focused)
     -0:11 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:11 remote-ftp:disconnect (ul.list-inline.tab-bar.inset-panel)
     -0:04.0 core:confirm (atom-text-editor.editor.mini.is-focused)
```
### Config

``` json
{
  "core": {
    "disabledPackages": [
      "linter-bootlint",
      "atom-htmltidy"
    ],
    "telemetryConsent": "limited"
  },
  "Remote-FTP": {
    "hideLocalWhenDisplayed": true
  }
}
```
### Installed Packages

``` coffee
# User
atom-beautify, v0.29.13 (inactive)
atom-htmltidy, v5.1.0 (inactive)
highlight-selected, v0.11.2 (active)
intentions, v1.1.2 (active)
linter, v1.11.18 (active)
linter-bootlint, v1.0.2 (inactive)
linter-csslint, v1.3.3 (active)
linter-flake8, v2.0.2 (active)
linter-jshint, v3.0.1 (active)
linter-jsonlint, v1.3.0 (active)
linter-tidy, v2.2.1 (active)
minimap, v4.25.4 (active)
minimap-linter, v1.6.0 (active)
open-html-in-browser, v0.1.0 (inactive)
pigments, v0.37.0 (active)
project-manager, v3.2.5 (active)
Remote-FTP, v0.9.0 (active)
split-diff, v1.0.5 (inactive)
atom-dark-syntax, v0.27.0 (inactive)
atom-dark-ui, v0.52.0 (inactive)
atom-light-syntax, v0.28.0 (inactive)
atom-light-ui, v0.44.0 (inactive)
base16-tomorrow-dark-theme, v1.2.0 (inactive)
base16-tomorrow-light-theme, v1.2.0 (inactive)
one-dark-ui, v1.6.0 (active)
one-light-ui, v1.6.0 (inactive)
one-dark-syntax, v1.3.0 (active)
one-light-syntax, v1.3.0 (inactive)
solarized-dark-syntax, v1.0.2 (inactive)
solarized-light-syntax, v1.0.2 (inactive)
about, v1.7.0 (active)
archive-view, v0.61.1 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.11.2 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.31.1 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.27.0 (inactive)
autosave, v0.23.1 (active)
background-tips, v0.26.1 (active)
bookmarks, v0.42.0 (active)
bracket-matcher, v0.82.1 (active)
command-palette, v0.38.0 (active)
deprecation-cop, v0.54.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.40.0 (active)
find-and-replace, v0.201.1 (inactive)
fuzzy-finder, v1.4.0 (active)
git-diff, v1.1.0 (active)
go-to-line, v0.31.0 (inactive)
grammar-selector, v0.48.2 (active)
image-view, v0.59.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.0 (active)
link, v0.31.1 (inactive)
markdown-preview, v0.158.0 (active)
metrics, v1.0.0 (active)
notifications, v0.65.1 (active)
open-on-github, v1.2.0 (inactive)
package-generator, v1.0.0 (inactive)
settings-view, v0.242.2-hotfix1 (active)
snippets, v1.0.2 (active)
spell-check, v0.68.2 (active)
status-bar, v1.4.1 (active)
styleguide, v0.47.0 (active)
symbols-view, v0.113.1 (inactive)
tabs, v0.101.0 (active)
timecop, v0.33.2 (active)
tree-view, v0.209.3 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.35.1 (active)
whitespace, v0.33.0 (active)
wrap-guide, v0.38.2 (active)
language-c, v0.52.1 (active)
language-clojure, v0.21.0 (active)
language-coffee-script, v0.47.2 (active)
language-csharp, v0.12.1 (active)
language-css, v0.37.1 (active)
language-gfm, v0.88.0 (active)
language-git, v0.15.0 (active)
language-go, v0.42.1 (active)
language-html, v0.45.1 (active)
language-hyperlink, v0.16.0 (active)
language-java, v0.23.0 (active)
language-javascript, v0.119.0 (active)
language-json, v0.18.2 (active)
language-less, v0.29.5 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.35.0 (active)
language-php, v0.37.2 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.0 (active)
language-ruby, v0.69.0 (active)
language-ruby-on-rails, v0.25.0 (active)
language-sass, v0.56.0 (active)
language-shellscript, v0.22.4 (active)
language-source, v0.9.0 (active)
language-sql, v0.23.0 (active)
language-text, v0.7.1 (active)
language-todo, v0.28.0 (active)
language-toml, v0.18.0 (active)
language-xml, v0.34.9 (active)
language-yaml, v0.26.0 (active)

# Dev
No dev packages
```

**Comment by jimmaaay** [comment]
Thanks for pointing this out :)

**Comment by brentpayton** [comment]
You're welcome!  Your plugin is essential to the way I use Atom.

**Comment by jpxd** [comment]
Should be fixed with 4f481dec3e4749bcd062edfab2b085e46d8e7d40 :)
